lib/rspec/buildkite/analytics/ci.rb

```diff
@@ -2,30 +2,83 @@
 
 require "securerandom"
 
-module RSpec::Buildkite::Analytics::CI
+class RSpec::Buildkite::Analytics::CI
   def self.env
-    if ENV["BUILDKITE_BUILD_ID"]
-      {
-        "CI" => "buildkite",
-        "key" => ENV["BUILDKITE_BUILD_ID"],
-        "url" => ENV["BUILDKITE_BUILD_URL"],
-        "branch" => ENV["BUILDKITE_BRANCH"],
-        "commit_sha" => ENV["BUILDKITE_COMMIT"],
-        "number" => ENV["BUILDKITE_BUILD_NUMBER"],
-        "job_id" => ENV["BUILDKITE_JOB_ID"],
-        "message" => ENV["BUILDKITE_MESSAGE"],
-        "debug" => ENV["BUILDKITE_ANALYTICS_DEBUG_ENABLED"],
-        "version" => RSpec::Buildkite::Analytics::VERSION,
-        "collector" => RSpec::Buildkite::Analytics::NAME
-      }
-    else
-      {
-        "CI" => nil,
-        "key" => SecureRandom.uuid,
-        "debug" => ENV["BUILDKITE_ANALYTICS_DEBUG_ENABLED"],
-        "version" => RSpec::Buildkite::Analytics::VERSION,
-        "collector" => RSpec::Buildkite::Analytics::NAME
-      }
-    end
+    new.env
+  end
+
+  def env
+    ci_env.merge(default)
+  end
+
+  private
+
+  def ci_env
+    return buildkite if ENV["BUILDKITE_BUILD_ID"]
+    return github_actions if ENV["GITHUB_RUN_NUMBER"]
+    return circleci if ENV["CIRCLE_BUILD_NUM"]
+    return generic if ENV["CI"]
+
+    {
+      "CI" => nil,
+      "key" => SecureRandom.uuid,
+    }
+  end
+
+  def default
+    {
+      "key" => ENV["BUILDKITE_ANALYTICS_KEY"],
+      "url" => ENV["BUILDKITE_ANALYTICS_URL"],
+      "branch" => ENV["BUILDKITE_ANALYTICS_BRANCH"],
+      "commit_sha" => ENV["BUILDKITE_ANALYTICS_SHA"],
+      "number" => ENV["BUILDKITE_ANALYTICS_NUMBER"],
+      "job_id" => ENV["BUILDKITE_ANALYTICS_JOB_ID"],
+      "message" => ENV["BUILDKITE_ANANLYTICS_MESSAGE"],
+      "debug" => ENV["BUILDKITE_ANALYTICS_DEBUG_ENABLED"],
+      "version" => RSpec::Buildkite::Analytics::VERSION,
+      "collector" => RSpec::Buildkite::Analytics::NAME,
+    }.reject { |k, v| v.nil? }
```

Comment:
`.compact` does the same thing? Is this an ActiveSupport thing again 🤔

Reply:
No I'm pretty sure I've used `.compact` in other places in this code base...

Reply:
Yep you can. TIL you can compact a hash. Nice!